include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -308,21 +292,15 @@ class IExecutionProvider {
     return nullptr;
   }
 
+  virtual OrtDevice GetOrtDeviceByMemType(OrtMemType mem_type) const;
+
+  virtual std::vector<AllocatorPtr> CreatePreferredAllocators() { return std::vector<AllocatorPtr>(); };
```

Comment:
>AllocatorPtr [](http://example.com/codeflow?start=22&length=12)

should we return unique_ptr instead of shared_ptr? as we are transfer the ownership